lib/net.c

```diff
@@ -542,6 +542,18 @@ get_ip_type(const char *ip_addr)
 	return ret;
 }
 
+bool
+ipv4_if_configured(struct gatekeeper_if *iface)
```

Comment:
Shouldn't this function be inline?

Bug: if `bool`'s size is 1 bit, the bitwise test below may not be safe. The solution is wrap the whole expression with `!!` to guarantee that the result is `bool`.


Shouldn't you replace the old code that uses GK_CONFIGURED_IPV4 directly?